### PR TITLE
Start charging after first month free & Cancelling should charge immediately.

### PR DESCRIPTION
**Pull request by cihangir**
## Description

We should start charging our users after their ~~trial~~ free period ends. So their presence info will be collected after first moth ends. During free month, set presence info's is processed as true.

On cancelling subscription, we should immediately charge and cancel the sub.
## Motivation and Context

See pricing SRS
## How Has This Been Tested?

Has unit tests
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by cihangir** [comment]
cc/ @usirin

**Comment by usirin** [comment]
@rjeczalik gentle ping.

File: go/src/socialapi/Makefile [hunk 1]
```diff
@@ -227,6 +227,12 @@ testwebhook:
 	@./tests.test $(DBG) $(EXTRAS) -c $(CONFIG)
 	@rm ./tests.test
 
+testpresence:
+	@echo "$(OK_COLOR)--> presence tests... $(NO_COLOR)"
+	@`which go` test -race -c workers/presence/*.go
```
Comment (rjeczalik) [inline go/src/socialapi/Makefile]: just `go test -race -c koding/workers/presence`

File: go/src/socialapi/workers/payment/api/subscription_test.go [hunk 2]
```diff
@@ -176,6 +176,62 @@ func TestSubscribingToPaidPlanWithWithDifferentTrialPeriodThanDefault(t *testing
 	})
 }
 
+func TestCancellingSubscriptionCreatesAnotherInvoice(t *testing.T) {
+	Convey("Given stub data", t, func() {
+		withTestServer(t, func(endpoint string) {
+			withStubData(endpoint, func(username, groupName, sessionID string) {
+				withTrialTestPlan(func(planID string) {
+					createURL := fmt.Sprintf("%s%s", endpoint, EndpointSubscriptionCreate)
```
Comment (rjeczalik) [inline go/src/socialapi/workers/payment/api/subscription_test.go]: nit-pick: can be just `endpoint + EndpointSubscriptionCreate` instead

File: go/src/socialapi/workers/presence/presence.go [hunk 3]
```diff
@@ -21,6 +25,11 @@ const (
 
 var (
 	pingCache = cache.NewMemoryWithTTL(time.Hour)
```
Comment (rjeczalik) [inline go/src/socialapi/workers/presence/presence.go]: you'd also need to call `pingCache.StartGC(time.Hour)` somewhere e.g. in init()

File: go/src/socialapi/workers/presence/presence.go [hunk 4]
```diff
@@ -21,6 +25,11 @@ const (
...
+
+	// this may lead to max 5 mins of invalid tracking data if a group changes
+	// their sub status during that period
+	groupCache = cache.NewMemoryWithTTL(time.Minute * 5)
```
Comment (rjeczalik) [inline go/src/socialapi/workers/presence/presence.go]: same here

**Review by rjeczalik — APPROVED** [review]

File: go/src/socialapi/Makefile (hunk 1 above)
Comment (cihangir) [inline go/src/socialapi/Makefile]: that was failing on wercker long time ago, if you check that makefile, all `go` command are `which go`

File: go/src/socialapi/workers/presence/presence.go (hunk 3 above)
Comment (cihangir) [inline go/src/socialapi/workers/presence/presence.go]: 👍  fixed

File: go/src/socialapi/Makefile (hunk 1 above)
Comment (rjeczalik) [inline go/src/socialapi/Makefile]: I mostly wanted to prefer package paths over file paths:

socialapi/workers/presence > workers/presence/*.go

Comment (cihangir) [inline go/src/socialapi/Makefile]: ah, ok

**Pull request merged**